src/main/java/com/google/devtools/build/lib/remote/RemoteSpawnCache.java

```diff
@@ -111,11 +111,11 @@
     this.topLevelOutputs = Preconditions.checkNotNull(topLevelOutputs, "topLevelOutputs");
   }
 
-
   @Override
   public CacheHandle lookup(Spawn spawn, SpawnExecutionContext context)
       throws InterruptedException, IOException, ExecException {
-    if (!Spawns.mayBeCached(spawn) || !Spawns.mayBeExecutedRemotely(spawn)) {
+    if (!Spawns.mayBeCached(spawn)
+        || (!Spawns.mayBeCachedRemotely(spawn) && !isDiskCache(options))) {
```

Comment:
`|| (!Spawns.mayBeCachedRemotely(spawn) && isRemoteCache(options)))`

wdyt? IMO, the intention will be easier to grasp.

Reply:
Sounds good :+1: